From 7248d3dc4519b2cf201b0f97d8802eafad062b0b Mon Sep 17 00:00:00 2001
From: Tien Hock Loh <tienhock.loh@starfivetech.com>
Date: Wed, 10 Aug 2022 06:01:53 -0400
Subject: [PATCH 48/68] third_party: libvpx: enable riscv64 build as generic

Added libvpx as dependency to nwjs and add libvpx
riscv64 build support.

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Signed-off-by: Tien Hock Loh <tienhock.loh@starfivetech.com>
---
 chrome/BUILD.gn             | 1 +
 third_party/libvpx/BUILD.gn | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 183e3b9b28c8..161410d3ef20 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1466,6 +1466,7 @@ group("dependencies") {
     "//services/tracing/public/cpp",
     "//third_party/blink/public:blink_devtools_frontend_resources",
     "//third_party/blink/public:blink_devtools_inspector_resources",
+    "//third_party/libvpx",
     "//v8:v8_headers",
   ]
 
diff --git a/third_party/libvpx/BUILD.gn b/third_party/libvpx/BUILD.gn
index 758aca154073..ee4de99375ac 100644
--- a/third_party/libvpx/BUILD.gn
+++ b/third_party/libvpx/BUILD.gn
@@ -35,6 +35,8 @@ if (current_cpu == "x86") {
   } else {
     cpu_arch_full = current_cpu
   }
+} else if (current_cpu == "riscv64") {
+  cpu_arch_full = "generic"
 } else {
   cpu_arch_full = current_cpu
 }
@@ -399,6 +401,9 @@ static_library("libvpx") {
       sources = libvpx_srcs_arm64
       public_deps = [ ":libvpx_arm64_headers" ]
     }
+  } else if (current_cpu == "riscv64") {
+    sources = libvpx_srcs_generic
+    public_deps = [ ":libvpx_generic_headers" ]
   }
 
   configs -= [ "//build/config/compiler:chromium_code" ]
-- 
2.30.2

